src/main/java/net/dries007/tfc/api/capability/player/PlayerDataHandler.java

```diff
@@ -0,0 +1,269 @@
+        {
+            nutrients[i] = 0.8f * MAX_PLAYER_NUTRIENTS;
+        }
+        for (int i = 0; i < skills.length; i++)
+        {
+            skills[i] = 0;
+        }
+        thirst = 70.0f;
+        exhaustion = 0;
+        deserializeNBT(nbt);
+    }
+
+    @Override
+    public float getNutrient(Nutrient nutrient)
+    {
+        update();
+        return nutrients[nutrient.ordinal()];
+    }
+
+    @Override
+    public float[] getNutrients()
+    {
+        update();
+        return nutrients;
+    }
+
+    @Override
+    public void setNutrients(float[] nutrients)
+    {
+        System.arraycopy(nutrients, 0, this.nutrients, 0, this.nutrients.length);
+    }
+
+    @Override
+    public void setNutrient(Nutrient nutrient, float amount)
+    {
+        if (amount < MIN_PLAYER_NUTRIENTS)
+        {
+            nutrients[nutrient.ordinal()] = MIN_PLAYER_NUTRIENTS;
+        }
+        else if (amount > MAX_PLAYER_NUTRIENTS)
+        {
+            nutrients[nutrient.ordinal()] = MAX_PLAYER_NUTRIENTS;
+        }
+        else
+        {
+            nutrients[nutrient.ordinal()] = amount;
+        }
+    }
+
+    @Override
+    public void addNutrient(Nutrient nutrient, float amount)
+    {
+        update();
+        float newAmount = nutrients[nutrient.ordinal()] + amount;
+        setNutrient(nutrient, newAmount);
+    }
+
+    @Override
+    public void update()
+    {
+        int ticksPassed = (int) (CalendarTFC.getCalendarTime() - lastUpdateTick);
+        for (Nutrient nutrient : Nutrient.values())
+        {
+            setNutrient(nutrient, nutrients[nutrient.ordinal()] - (float) (ConfigTFC.GENERAL.playerNutritionDecayModifier * nutrient.getDecayModifier() * ticksPassed));
+        }
+        //Reduces thirst bar for normal living
+        thirst -= (float) (ConfigTFC.GENERAL.playerThirstModifier * ticksPassed / 240);
+        if (exhaustion > 0)
+        {
+            float ticksExhaustion = exhaustion * 20.0f; //1.0F of exhaustion per sec
+            if (ticksExhaustion < ticksPassed)
+            {
+                exhaustion = 0;
+            }
+            else
+            {
+                exhaustion -= ticksPassed / 20f;
+                ticksExhaustion = ticksPassed;
+            }
+            thirst -= 0.025f * ticksExhaustion;
+        }
+        if (thirst < 0) thirst = 0;
+        lastUpdateTick = CalendarTFC.getCalendarTime();
+    }
+
+    @Override
+    public float getMaxHealth()
+    {
+        float nutrientPercent = getNutrientAverage() / MAX_PLAYER_NUTRIENTS;
+        float skillPercent = getSkillAverage() / 100; //MAX_PLAYER_SKILLS; This should return a value between 0 and 1.0F
+        float averagePercent = (nutrientPercent + skillPercent) / 2.0f;
+        float maxHealth;
+        if (averagePercent < BASE_HEALTH_THRESHOULD)
+        {
+            //Less than base health
+            if (averagePercent <= MIN_HEALTH_THRESHOULD) return ConfigTFC.GENERAL.playerMinHealth;
```

Comment:
This should be modifier based - no config references, no health amount references.

Reply:
Changed to use modifiers instead.